priv/posts/performance-methodologies.md

```diff
@@ -0,0 +1,170 @@
+
+> The foundation of your software should be resilient to achieve better results.
+
+## Monitoring
+
+Today, a big part of the market is going to follow distributed systems, as we know, that adds a lot of complexity to your architecture in exchange for scalability (resilience).
+
+Also, it adds more components as dependency or dependents. Therefore, you should monitor these components to have better visibility when the world is burning.
+
+We need to monitor every part of Software and Architecture, which will help when we go back to historical metrics to answer some of these questions:
+When the software performance is decreasing?
+When is the peak of the software access?
+How our devices are working on a specific date? Like I/O latency, DNS resolution
+These questions will help to choose the right performance methodology to apply.
+
+## Known-Unknowns
+
+![Known Unknowns](/images/performance-analysis/diagram-known-unknowns.png)
+
+> _This section is a reference to [System Performance](https://www.goodreads.com/book/show/18058001-systems-performance) book by Brendan Gregg._
+
+In performance analysis we can split information into three types:
+Know-Knows: These are things you know, for instance, you know that you should be checking CPU utilization **and** you know that the value is 10% on average.
+Know-Unknows: There are things that you know that you do not know. You know that an increase in API response time can be related to a third-party component, but you don't have metrics showing it.
+Unknown-Unknowns: These are things you do not know you do not know.  Confused? Let me elaborate, you may not know that DNS resolution can become heavy I/O latency, so you are not checking them (because you don't know).
+
+On architecture diagram creation is obvious to say that _unknowns-unknowns_  is not mappable, after all, you don't know.
+
+> Usually there is a lot of unknown-unknows and at least, is your job as a researcher in this field transform the `unknown-unknowns` into `know-unknows`.
+
+The Diagram above map `known-knowns` (Green box) and `known-unknows` (Red box)
+
+## Observability Tools
+
+As previously mentioned, has observability in our software/architecture is fundamental to perform performance improvements. In this section, I'll briefly show you great tools to use in applications.
+
+### Tracing
+
+Tracing collects per-event data for analysis. Normally, Tracing tools are not enabled by default, since it adds CPU overhead to capture the data.
+
+Logging (include system log) can be categorized as low-frequency tracing that is enabled by default.
+
+Some common tools:
+
+**system-wide**:
+- `tcpdump`: network packet tracing
+- `perf`: Linux Performance Events (tracing static and dynamic probes)
+
+**per-process**:
+- `strace`: system call tracing
+- `USDT` (Userland Statically Defined Tracing)
+- `DTrace`: observability framework that includes a programming language and a tool.
+
```

Comment:
I feel you're missing the "web" stack of tracing here. Jaeger, opentelemetry, zipkin, ..